src/components/ApiAction/QueryList.vue

```diff
@@ -18,33 +15,35 @@
               </b-col>
             </b-row>
             <template v-else>
-              <b-list-group class="leftNav-container" ref="leftNav-container">
-                <b-list-group-item
-                  v-for="query of paginatedQueries"
-                  class="px-3 py-0"
-                  :key="`saved-query-${query.idx}`"
-                  :active="query.idx === currentQueryIndex"
-                  :ref="`saved-query-${query.idx}`"
-                  :data-cy="`api-actions-saved-query-${query.name}`"
-                >
-                  <b-row align-v="center">
-                    <b-col
-                      cols="9"
-                      class="text-left py-3 pointer leftTab"
-                      @click="loadSavedQuery(query.idx)"
-                      :id="`query-list-${query.idx}`"
-                    >
-                      <span>{{ query.name }}</span>
-                    </b-col>
-                    <b-col cols="3" class="py-3">
-                      <i
-                        class="fas fa-trash pointer"
-                        @click="deleteSavedQuery(query)"
-                      />
-                    </b-col>
-                  </b-row>
-                </b-list-group-item>
-              </b-list-group>
+              <b-card no-body header="Saved API Actions">
+                <b-list-group class="leftNav-container" ref="leftNav-container">
+                  <b-list-group-item
+                    v-for="query of paginatedQueries"
+                    class="px-3 py-0"
+                    :key="`saved-query-${query.idx}`"
```

Comment:
```suggestion
                    :key="`saved-query-${query.name}`"
```
Using the index of the array element doesn't help the VDom library to resolve which element should be re-rendered and how, since indexes change when the array is mutated. Use instead more business-related properties, like the name or some property that really differentiates one element from the others.

Reply:
Actually, with a localStorage manipulation we could have two queries with the same name, I used idx to avoid key duplicate and so errors

Reply:
That's the difficult part about item lists with frameworks based on vDom. You should really find a way to create a string that uniquely identifies each item. And if you have items that are not unique (meaning they have the same exact content), then it's not a problem for the vDom, because it means the item doesn't need to be re-rendered.

Reply:
In the app we don't allow duplicates names for API Actions, the only way to there could be a duplicate is if an user manipulate directly the local storage
Should we handle this too ? What should be the behavior in that case ?
In case of a duplicate which API Action should we hide ? If we do that the user won't be able to save it with another name
At least we could just check if there are duplicates, hide them and display a warning toast to tell the user check the localStorage ?

Reply:
no no, I don't think it's a good idea to allow to save duplicate actions, I was only talking about it in general terms

Reply:
I wasn't talking about allowing save duplicate but how to handle the case when a user add duplicate directly in localStorage

Reply:
oh, ok

Reply:
I think it's a good idea to add a duplicate-management mechanism. I'll try to figure something out to prevent duplicates to land on the action list.

Reply:
In the meanwhile, I'd prefer to use the action name as the v-for key, which looks safe enough.